### DISCUSSION TITLE
Multiple Values in Message

### DISCUSSION THREAD

**Discussion by thelosmos** [arachnetech/homebridge-mqttthing#577] [Q&A] [answered]
Does anyone know how I would parse an MQTT message that contains more than one value? I am trying to configure the Get On topic so that the state of a light switch stays between homebridge/homekit and if its manually flipped.

It is a GE/Jasco ZWave switch. The topic contains a time stamp and a value. I just need the value.

Topic: zwave/Front_Door_Light/37/0/currentValue

Return: {"time":1659546409405,"value":false}

**Comment by mrMiimo** [comment]
Same for me

**Comment by HolgerAusB** [comment, marked as answer]
I can't test this but after reading other issues that contains "JSON" I think you could try to edit the config by hand and add an 'apply'-line

```
"topics": {
   "getOn": {
      "topic": "zwave/Front_Door_Light/37/0/currentValue",
      "apply": "return JSON.parse(message).value"
   },
...
}

**Reply by thelosmos** [reply]
That worked! Thank you @HolgerAusB! Posting my exact code for anyone that needs it:
```
    "topics": {
        "getOnline": "zwave/Front_Door_Light/status",
        "getOn": {
            "topic": "zwave/Front_Door_Light/37/0/currentValue",
            "apply": "return JSON.parse(message).value"
        },
        "setOn": "zwave/Front_Door_Light/37/0/targetValue/set"
    },
```

**Comment by thelosmos** [comment]
@HolgerAusB I believe I found a bug/issue. If I make this change via the JSON config, and later view the config in the UI, this config disappears.

**Reply by HolgerAusB** [reply]
As I wrote, I just saw a possible solution while searching the existing issues here in the repo. I newer needed or tested this. So If you think you found a new bug, you should open a new issue.
